### ISSUE TITLE
DMS pricing calculated on wrong region

### ISSUE THREAD

**Issue by jessedobbelaere** [infracost/infracost#177]
I see support for DMS is added in #149 🙌  

When running `infracost v0.6.4` I see that DMS cost is calculated at $0.0720/hour, however, these resources run in `eu-central-1` which have a different price. 

* Price shown by infracost for `t2.small` (multi-az): `$0.072/hour` --> probably corresponds to price fetched from `us-east-1`?
* Price for `t2.small` (multi-az)  in `eu-central-1`: `$0.084/hour` (According to: https://aws.amazon.com/dms/pricing/)

![image](https://user-images.githubusercontent.com/1352979/95872994-c0140300-0d6f-11eb-83b9-ed20120e3447.png)

However, I see `region` support in the `dms.go`, but my golang knowledge is zero to none to be able to debug it properly.

**Comment by sinabakh** [comment]
Hi, that's a bit strange. Thanks for the complete description of the problem. I will take a look at it ASAP.

**Comment by alikhajeh1** [comment]
@jessedobbelaere I think I spoke too soon, that fix looks wrong, your issue mentions multi-az so I'll keep digging...

**Comment by sinabakh** [comment]
@alikhajeh1 We already manage the multi-az flag, there was a bug in region selection. I think all is ok however we can till confirmed.

**Comment by alikhajeh1** [comment]
@sinabakh according to https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dms_replication_instance#availability_zone the availability_zone is optional, and that field cannot be set if multi_az is set (which is what the issue description mentions). I might be missing something?

**Comment by sinabakh** [comment]
@alikhajeh1 Right, the pricing page shows different prices for different regions even for multi-az. And based on my tests, I could set the `availability_zone` alongside the `multi-az`. I might be missig sth so it would be great if you can look through the code and logics.

**Comment by alikhajeh1** [comment]
@sinabakh Terraform doesn't validate all parameters, it passes them to AWS and returns an error if AWS fails to launch, so we can only tell for sure if we try to launch a DMS. That can get tricky so I thought we could just wait for @jessedobbelaere to share an example resource that lets us reproduce the issue.

**Comment by alikhajeh1** [comment]
Thanks @jessedobbelaere. Infracost selects the region based on the provider config, we're adding support for https://www.terraform.io/docs/configuration/providers.html#alias-multiple-provider-configurations just now, is that how you were setting the region for this resource? If not, can you please let us know how the region is being set?

Also, can you please confirm if `replication_instance_arn` has "eu-central-1" in it? We're currently falling back to extracting the region from `arn` if we can't get it from the provider config, but for some reason, `aws_dms_replication_instance` calls it `replication_instance_arn`. We think we can remove region-detection-from-arn when we support aliases and handle any other methods that are used to set the region.

**Comment by jessedobbelaere** [comment]
Ok 👍  At the moment we only have 1 aws provider which has the `eu-central-1` regio defined though. The ARN of my DMS instance:

```
"replication_instance_arn": "arn:aws:dms:eu-central-1:XXX:rep:XXX",
```

**Comment by alikhajeh1** [comment]
> aws provider which has the eu-central-1 regio defined though

Using this?
```
provider "aws" {
  region                      = "eu-central-1"
  ...
}
```

**Comment by jessedobbelaere** [comment]
Oh I see the issue in my terraform plan

```
  "configuration": {
    "provider_config": {
      "aws": {
        "name": "aws",
        "expressions": { "region": { "references": ["var.region"] } }
      },
```

In our CI/CD pipeline, we have an env var that lists our regions, and we loop over them and run terraform plan/apply for each region

```
terraform plan -input=false -var "region=${REGION}" -out=${CI_ENVIRONMENT_NAME}_${REGION}.plan
```

**Comment by sinabakh** [comment]
So I created another issue to address this problem
https://github.com/infracost/infracost/issues/183

**Comment by sinabakh** [comment]
Now #183 is merged into master I think this issue is resolved too. @alikhajeh1 ping us here when a new release is available so @jessedobbelaere can check it.

**Comment by aliscott** [comment]
Thanks @sinabakh this is released in v0.6.5.
Thanks @jessedobbelaere please reopen this issue if your problem is not fixed.

**Comment by jessedobbelaere** [comment]
Tested and it has the correct DMS pricing now for the region ! Thanks for the super quick fix 🙌 

![image](https://user-images.githubusercontent.com/1352979/96291447-52264080-0fe8-11eb-8ad2-5d651621e5a5.png)
